contentcuration/contentcuration/constants/completion_criteria.py

```diff
@@ -14,10 +31,9 @@ def validate(data, kind=None):
     if isinstance(data, (dict,)) and not data:
         return
 
-    try:
-        jsonschema.validate(instance=data, schema=completion_criteria.SCHEMA)
-    except jsonschema.ValidationError as e:
-        raise ValidationError("Completion criteria does not conform to schema") from e
+    error = best_match(validator.iter_errors(data))
+    if error is not None:
+        raise ValidationError("Completion criteria does not conform to schema") from error
```

Comment:
Could we add more details from the error here, rather than this more generic response?

Reply:
Is the `from error` sufficient? I believe Python and Sentry will capture the originating error message. I'm afraid though that the message isn't always helpful, for instance:
```python
In [5]: completion_criteria.validate({"model": "does not exist"}, kind=content_kinds.DOCUMENT)
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
ValidationError: 'time' was expected

Failed validating 'const' in schema[0]:
    {'const': 'time'}

On instance:
    'does not exist'

The above exception was the direct cause of the following exception:

ValidationError                           Traceback (most recent call last)
<ipython-input-5-d4d2b780619e> in <module>
----> 1 completion_criteria.validate({"model": "does not exist"}, kind=content_kinds.DOCUMENT)

~/Projects/learningequality/studio2/contentcuration/contentcuration/constants/completion_criteria.py in validate(data, kind)
     34     error = best_match(validator.iter_errors(data))
     35     if error is not None:
---> 36         raise ValidationError("Completion criteria does not conform to schema") from error
     37 
     38     model = data.get("model")

ValidationError: ['Completion criteria does not conform to schema']

```

Reply:
I took `best_match(validator.iter_errors(data))` from the `jsonschema` source, but perhaps it'd be better to summarize the errors ourselves instead of trying to do that

Reply:
Sorry, I had totally missed the `from` clause here - but yeah, we might be better off doing some additional processing of the json schema error to make it more parseable.